### PR TITLE
feat: Impl serialization

### PR DESCRIPTION
**Pull request by shaojunda**

### CODE REVIEW

File: lib/ckb/api.rb [hunk 1]
```diff
@@ -37,7 +37,7 @@ def initialize(host: CKB::RPC::DEFAULT_URL, mode: MODE::TESTNET)
           index: "0"
         )
 
-        secp_cell_type_hash = rpc.compute_script_hash(system_cell_transaction.outputs[1].type.to_h)
+        secp_cell_type_hash = system_cell_transaction.outputs[1].type.to_hash
```
Comment (ashchan) [inline lib/ckb/api.rb]: We have `to_h` & `from_h` for these primitive types. Along with that, `to_hash` & `hash` are similar names with very different meanings. This is especially obvious for Ruby as `Hash` is a fundamental type.

I suggest that we rename `Script#to_hash` to `Script#serialize` to make it more explicit.

Comment (shaojunda) [reply lib/ckb/api.rb]: thanks, it make sense. I will change that.

Comment (shaojunda) [reply lib/ckb/api.rb]: but for `Script`, serialize should return something like `5c010000140000005001000054010000580100003c0100000c000000`,`to_hash` need return Script's hash. so will `calc_script_hash` be better?

Comment (ashchan) [reply lib/ckb/api.rb]: Interesting. Would it be worth making it further explicit like `script_hash` (and `tx_hash` for `Transaction`). A bit redundant but won't make much confusion.

Comment (ashchan) [reply lib/ckb/api.rb]: These interfaces will be used widely. Naming them without confusion is more important than making them short.

Comment (shaojunda) [reply lib/ckb/api.rb]: `script_hash` is quite suitable, but `tx_hash` is too similar to `hash`(Transaction's filed)?

Comment (ashchan) [reply lib/ckb/api.rb]: On a second thought, we can just use `hash` as property/method, but use `compute_hash` to do the serialize and calculation.

Comment (shaojunda) [reply lib/ckb/api.rb]: > On a second thought, we can just use `hash` as property/method, but use `compute_hash` to do the serialize and calculation.

thanks, great, I will change it.

**Pull request merged**